src/pages/index.js

```diff
@@ -0,0 +1,361 @@
+    .cloneNode(true);
+
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const cardLikeBtn = cardElement.querySelector(".card__like-button");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-button");
+
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-button-liked");
+  }
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.setAttribute("src", data.link);
+  cardImageEl.setAttribute("alt", data.name);
+  // Can also be written as:
+  // cardImageEl.src = data.link
+  // cardImageEl.alt = data.alt
+
+  cardLikeBtn.addEventListener("click", (evt) => handleLike(evt, data._id));
+
+  cardDeleteBtn.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data._id)
+  );
+
+  cardImageEl.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+    previewModalCaptionEl.textContent = data.name;
+  });
+
+  return cardElement;
+}
+
+previewCloseBtn.addEventListener("click", () => {
+  closeModal(previewModal);
+});
+
+function handleEscKey(evt) {
+  if (evt.key === "Escape") {
+    const openedModal = document.querySelector(".modal_opened");
+    closeModal(openedModal);
+  }
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscKey);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscKey);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+
+  const cardSubmitBtn = evt.submitter;
+  setButtonText(cardSubmitBtn, true, "Saving...");
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(cardSubmitBtn, false, "Save");
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+  const avatarSubmitBtn = evt.submitter;
+  setButtonText(avatarSubmitBtn, true, "Saving...");
+
+  api
+    .editAvatarInfo(avatarLinkInput.value)
+    .then((data) => {
+      avatar.src = data.avatar;
+      closeModal(avatarModal);
+      evt.target.reset();
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(avatarSubmitBtn, false, "Save");
+    });
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+
+  const addSubmitBtn = evt.submitter;
+  setButtonText(addSubmitBtn, true, "Saving...");
+
+  api
+    .addNewCard({
+      name: cardNameInput.value,
+      link: cardLinkInput.value,
+    })
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      evt.target.reset();
```

Comment:
* You need to **`disable`** the submit button in the `card form` if the inputs are **empty**. It’s needed because I can click the button and add an empty card, but it should **not** happen. After the first submission, the button gets activated, and you need to **deactivate** it again. Even if I can't add an empty card, the button should look **disabled** (gray) anyway and have a `disabled` attribute.